integ/emf_logger/logscript.py

```diff
@@ -0,0 +1,44 @@
+import time
+import sys
+import os
+import random
+import socket
+
+"""
+This script writes a single EMF payload to the TCP socket
+https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Embedded_Metric_Format_Specification.html
+"""
+
+# Define constants
+metricNamespace = "fluent-metrics"
+metricName = 'fluent-bit-integ-test-{:d}'.format(random.randint(0, 32767))
+metricValue = 100
+
+# pass the metric-name to a file in the shared volume for validation
+metricPath = os.getenv('EMF_METRIC_NAME_PATH')
+file = open(metricPath, "w")
+file.write(metricName)
+file.close()
+
+# EMF payload.
+emfPayload = ('{{"_aws":{{"Timestamp":{:d}000,"CloudWatchMetrics":[{{"Namespace":"{:s}","Dimensions":[["dimensionKey"]]'
+              ',"Metrics":[{{"Name":"{:s}"}}]}}]}},"dimensionKey":"dimensionValue","{:s}":{:d}}}\n').format(
+    int(time.time()), metricNamespace, metricName, metricName, metricValue)
+
+dataToSend = str.encode(emfPayload)
+
+# On Linux, connect to the Unix socket.
+# For Windows, we need to connect to a TCP socket running in Fluent-bit container.
```

Comment:
I'm confused by this. I dont see any code that handles the Linux case here?

Reply:
EMF should always work over the TCP socket, connecting to the unix socket should not be possible as it is mounted into the firelens container only by the ECS Agent from a host path that it chooses...

https://github.com/aws-samples/amazon-ecs-firelens-examples/tree/mainline/examples/fluent-bit/cloudwatchlogs-emf

> I'm confused by this. I dont see any code that handles the Linux case here?

I'm really confused by this comment as well

Reply:
So, I have updated this segment along with the comments in the newer revision. 
This was inaccurate in the previous revision.

Basically, we would connect over TCP only. To resolve the IP address, we can either use service name or the IP.